tools/system_libs.py

```diff
@@ -614,11 +614,9 @@ def do_create():
       force = force.union(deps)
   ret.sort(key=lambda x: x.endswith('.a')) # make sure to put .a files at the end.
 
-  # Handle backend compiler_rt separately because it is not a bitcode system lib like the others.
```

Comment:
Ah, this is why it was handled differently before. Previously compiler_rt.a was built as archives of .s files, and needed to be manually passed in to s2wasm to do .s linking. The switch to LLD made it so that we build compiler_rt as an archive of .o files, and then link with LLD. The final change that makes this patch possible is removing llvm-link from the build process, and just sending files to LLD, which is able to handle a mixture of .bc and .o files.